content/blog/wallet-descriptors.md

```diff
@@ -0,0 +1,164 @@
+
+## Wallet descriptors in Bitcoin Core
+
+Core has had support for output descriptors since v17.0. In this section we are going to explore 
+a few Bitcoin Core commands with support for output descriptors. From Bitcoin Core v23.0.0, if you create a wallet with the CLI you will get a descriptor wallet unless `descriptors=false` is set 
+while using `createwallet` to create the wallet. Assuming you already have a wallet, we will start
+by creating an address, I'm running v23.0.0 but any version above v17.0 should be okay.
+
+Run the following command to create an address:
+`bitcoin-cli -rpcwallet=<wallet-name> getnewaddress`
+
+This should get you a bech32 bitcoin address. Using `getaddressinfo` let's see what Core offers us 
+apart from the address.
+
+`bitcoin-cli -rpcwallet=<wallet-name> getaddressinfo <previous-generated-address>`
+
+After running this command you should have a JSON object as result with a `desc` which contains your output descriptor. It should look similar to this:
+
+```json
+{
+    "address": "bcrt1q2e4g8cetupxes7l95qstpxq8qnpml9ev6gjle2",
+    "scriptPubKey": "0014566a83e32be04d987be5a020b0980704c3bf972c",
+    "ismine": true,
+    "solvable": true,
+    "desc": "wpkh([dc711bf8/0'/0'/1']0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9)#yr975l8h",
+    "iswatchonly": false,
+    "isscript": false,
+    "iswitness": true,
+    "witness_version": 0,
+    "witness_program": "566a83e32be04d987be5a020b0980704c3bf972c",
+    "pubkey": "0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9",
+    "ischange": false,
+    "timestamp": 1654093495,
+    "hdkeypath": "m/0'/0'/1'",
+    "hdseedid": "50b54a741dbc3b120280391ecba28d657e137bce",
+    "hdmasterfingerprint": "dc711bf8",
+    "labels": [""]
+}
+```
+
+We can get more information about the descriptor in the result above by running the following command:
+
+`bitcoin-cli getdescriptorinfo <descriptor>`
+
+Using the descriptor in the JSON output above, I get the following result.
+
+```json
+{
+  "descriptor": "wpkh([dc711bf8/0'/0'/1']0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9)#yr975l8h",
+  "checksum": "yr975l8h",
+  "isrange": false,
+  "issolvable": true,
+  "hasprivatekeys": false
+}
+```
+
+A script is solvable according to Core, if given the private key and the descriptor, the wallet software can generate a scriptsig/witness to spend its funds. `isrange` denotes if it is a range descriptor meaning it is possible to generate other descriptors from it. Range descriptors usually end their derivation path with `/*`. For example, `wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/0'/0'/0/*)` is a range descriptor.
```

Comment:
`hasprivatekeys` shows whether the descriptor itself contains a private key. Here the descriptor has only a pubkey, even though the underlying wallet *does* have the associated private key